delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/context/ContextValue.scala

```diff
@@ -24,12 +24,12 @@ final case class ContextValue private[jsonld] (value: Json) {
     */
   def merge(that: ContextValue): ContextValue =
     (value.asArray, that.value.asArray, value.asString, that.value.asString) match {
-      case (Some(arr), Some(thatArr), _, _) => arrOrObj(removeEmpty(arr ++ thatArr))
-      case (_, Some(thatArr), _, _)         => arrOrObj(removeEmpty(value +: thatArr))
-      case (Some(arr), _, _, _)             => arrOrObj(removeEmpty(arr :+ that.value))
-      case (_, _, Some(str), Some(thatStr)) => arrOrObj(removeEmpty(Seq(str.asJson, thatStr.asJson)))
-      case (_, _, Some(str), _)             => arrOrObj(removeEmpty(Seq(str.asJson, that.value)))
-      case (_, _, _, Some(thatStr))         => arrOrObj(removeEmpty(Seq(value, thatStr.asJson)))
+      case (Some(arr), Some(thatArr), _, _) => arrOrObj(removeEmptyAndDup(arr ++ thatArr))
```

Comment:
Making the ContextValue with an ADT with:
```
case object EmptyContext
case class   ContextList 
```
would make it easier no ?
We only really need json on the contextObj method, for the rest, it seems bring to be bothersome as json is something a little large

Reply:
I would not touch this part of the logic now, since it is sensitive. A `ContextValue` can be (from the point of view of Json)
- String
- Empty
- Object
- List

Reply:
Yep, I understand this is sensitive and could be done later, no worries.
The String and Object cases can be expressed as the List case to simplify things no ?
That is why I had only two cases in the ADT

Reply:
But then you'll need a list with another ADT on it with 3 cases.